packages/rulesets/src/spectral/az-arm.ts

```diff
@@ -790,6 +790,8 @@ const ruleset: any = {
       description: "This rule ensures that the authors explicitly define these restrictions as a regex on the resource name.",
       message: "{{error}}",
       severity: "error",
+      disableForTypeSpec: true,
+      disableForTypeSpecReason: "Covered by TSP's '@azure-tools/typespec-azure-resource-manager/arm-resource-name-pattern' rule.",
```

Comment:
@tjprescott you also must bump the package version (PATCH) and run some `rush` commands and then deploy (I will have to approve). Details here:
https://github.com/Azure/azure-openapi-validator/blob/main/CONTRIBUTING.md#deploy-to-prod-lintdiff

Because this is only rule PATCH version, there should be no need to bump anything in `openapi-alps` so you don't have to stress too much.

Also, important: you don't really have to run `rush change` or bump any other package versions. The instructions are outdated - I simplified the process since then but didn't have time to update the docs yet.

Reply:
https://github.com/Azure/azure-openapi-validator/issues/683